mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -37,50 +37,98 @@ use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
 use crate::mm2::lp_swap::{active_swaps_using_coin, tx_helper_topic, watcher_topic};
 use crate::mm2::MmVersionResult;
 
+const INTERNAL_SERVER_ERROR_CODE: u16 = 500;
+
+/// Get enabled `platform coin` tokens if `mm2` receives a request to disable a `platform token`.
+async fn get_enabled_platform_coin_tokens(ctx: &MmArc, coin: MmCoinEnum) -> Result<Vec<String>, String> {
+    let ctx = CoinsContext::from_ctx(ctx).map_err(|err| ERRL!("{}", err))?;
+    let get_tokens = |ticker| async move {
+        ctx.get_platform_coin_tokens(ticker)
+            .await
+            .map_err(|err| ERRL!("{:?}", err))
+    };
+
+    match coin {
+        MmCoinEnum::SlpToken(slp) => Ok(get_tokens(slp.platform_ticker()).await?),
+        #[cfg(all(not(target_os = "ios"), not(target_os = "android"), not(target_arch = "wasm32")))]
+        MmCoinEnum::SplToken(spl) => Ok(get_tokens(spl.platform_coin.ticker()).await?),
+        #[cfg(not(target_arch = "wasm32"))]
+        MmCoinEnum::LightningCoin(lightning) => Ok(get_tokens(lightning.platform.coin.platform_ticker()).await?),
+        MmCoinEnum::Bch(ref bch) => Ok(get_tokens(bch.platform_ticker()).await?),
+        _ => Ok(vec![]),
+    }
```

Comment:
Is there a reason to match for certain `MmCoinEnum` variants here? I see that `ETH` is not covered for instance and it can have `ERC20` tokens enabled. If there is no reason for such match, I think this block can be simplified alot this way
``` rust
    let platform_ticker = coin.platform_ticker();
    coin_ctx.get_platform_coin_tokens(platform_ticker)
        .await
        .map_err(|err| ERRL!("{:?}", err))
```

Reply:
I see now that if this is done the way I suggested, disabling one token will disable all the platform tokens with it. Maybe you can add a simple check `if platform_ticker == coin.ticker()` and return an error if this check is not `true`? I guess this check can be better added in `disable_coin`, so this function can be used with any token in other places. What do you think @borngraced ?

Reply:
thank you! this is a great suggestion and it works well